lxd/instance/instance_utils.go

```diff
@@ -128,6 +128,42 @@ func ValidConfig(sysOS *sys.OS, config map[string]string, expanded bool, instanc
 }
 
 func validConfigKey(os *sys.OS, key string, value string, instanceType instancetype.Type) error {
+	if !strings.HasPrefix(key, instancetype.ConfigVolatilePrefix) {
```

Comment:
I'm curious why `volatile.*` are special and not say `image.*`?

Reply:
For `volatile.*` keys, we check suffixes. For example:

https://github.com/canonical/lxd/blob/ac7aa741dc57cd7eb0a07dd7b5a66b9658a6e760/lxd/instance/instancetype/instance.go#L1139-L1159

Reply:
So we don't want to rule them as invalid, but rather pass them to `ConfigKeyChecker`.